### PR TITLE
Add git_branch_is_name_valid and git_tag_is_name_valid

### PR DESCRIPTION
**Pull request by csware**
Add separate function to validate branch name.

Cross-ref: issue #5506

### CODE REVIEW

File: include/git2/tag.h [hunk 1]
```diff
@@ -365,6 +365,16 @@ GIT_EXTERN(int) git_tag_peel(
  */
 GIT_EXTERN(int) git_tag_dup(git_tag **out, git_tag *source);
 
+/**
+ * Determine whether a tag name is valid
+ *
+ * @param name A tag name to test
+ *
+ * @return 1 if tag name not valid, 0 if it is valid, an error code otherwise.
```
Comment (ethomson) [inline include/git2/tag.h:373]: `is_name_valid` is a boolean question.  We should return something truthy if it's true, and false if it's false.  

`if (!git_branch_is_name_valid()) { name is valid }` doesn't make sense.  Further, this matches `git_reference_is_name_valid`.

Comment (csware) [reply include/git2/tag.h]: Updated.

**Comment by ethomson** [comment]
Thanks @csware - your pull request highlighted a problem with the existing `is_name_valid` functions, that they cannot communicate a failure to the user, and it's conflated with an invalid name.  I opened #5659 to deprecate the `is_name_valid` functions and introduce `name_is_valid` functions.

I've included these changes over there, modified to be able to communicate failure to the user, use the new naming conventions, and follow our code formatting and style rules.

**Comment by ethomson** [comment]
Included in #5659 - thanks again.

**Pull request closed without merging**